src/platform-includes/performance/enable-manual-instrumentation/go.mdx

```diff
@@ -1,14 +1,28 @@
 To instrument certain regions of your code, you can create transactions to capture them.
 
-The following example creates a transaction span to time runs of an expensive operation on items from a channel. Timing for each operation is sent to Sentry and grouped by transaction name:
+The following example creates a transaction based on an incoming request:
 
 ```go
-ctx := context.Background()
-
-for item := range ch {
-	span := sentry.StartSpan(ctx, "doWork",
-		sentry.WithTransactionName(fmt.Sprintf("doWork: %s", item.Type)))
-	doWork(span.Context(), item) // doWork may create additional spans
-	span.Finish()
-}
+http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	hub := sentry.GetHubFromContext(ctx)
+	if hub == nil {
+		hub = sentry.CurrentHub().Clone()
```

Comment:
This example uses a couple concurrency-related patterns (whose intent might not be immediately clear to users), and also uses HTTP-related example (for that people should probably use web-related integrations).

If you really want to have this example instead, let's maybe add a comment (e.g. pointing at https://docs.sentry.io/platforms/go/concurrency/) and also a note that if "you're doing HTTP stuff, then look at the integrations".

Reply:
We could also use a more straightforward example, though I would argue that the 90% use-case is likely an HTTP request.

Reply:
```suggestion
		// Check the concurrency guide for more details: https://docs.sentry.io/platforms/go/concurrency/
		hub = sentry.CurrentHub().Clone()
```